tests/dataset_manager/test_artifact.py

```diff
@@ -264,6 +264,39 @@ def test_loading_key_leaves_filters_unchanged():
         assert a.filter_terms == filter_terms
 
 
+def test_replace(hdf_mock):
+    path = '/place/with/artifact.hdf'
+    filter_terms = ['location == Global', 'draw == 10']
+    key = 'new.key'
+    ekey = EntityKey(key)
+
+    a = Artifact(path, filter_terms)
+
+    assert ekey not in a.keys
+
+    a.write(key, "data")
+
+    a.replace(key, "new_data")
+    hdf_mock.remove.assert_called_with(path, ekey)
+    hdf_mock.write.assert_called_with(path, ekey, "new_data")
```

Comment:
You should check for all calls: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_has_calls